code/modules/status_system/statusEffects.dm

```diff
@@ -87,16 +87,24 @@
 		 *
 		 * 	Required: sweatReagent - the chemical you're sweating
 		 *  targetTurf should be left default
+		 *  Turning sweatpools on causes sweat chempools instead of cleanables
 		 */
-	proc/dropSweat(var/sweatReagent, var/sweatAmount = 5, var/sweatChance = 5, var/turf/targetTurf = get_turf(owner))
+	proc/dropSweat(var/datum/reagent/sweatReagent, var/sweatAmount = 5, var/sweatChance = 5, var/turf/targetTurf = get_turf(owner), var/sweatpools = FALSE)
+		if (!prob(sweatChance))
+			return
 		var/datum/reagents/tempHolder = new
-		if (prob(sweatChance))
-			tempHolder.add_reagent(sweatReagent, sweatAmount)
-			targetTurf.fluid_react_single(sweatReagent,sweatAmount)
+		if (sweatpools)
+			tempHolder.add_reagent(sweatReagent.id, sweatAmount)
```

Comment:
This won't work, you're trying to access a variable of a type rather than an instance. Generally we prefer passing reagent IDs around (as it was before), because getting instances is a faff and requires you to go via the cache.